### ISSUE TITLE
[BUG] Only blank screen in UI Mode

### ISSUE THREAD

**Issue by Chrisman94** [microsoft/playwright#28364]
### System info
- Playwright Version: v1.40
- Operating System: macOS 13.2
- Browser: All
- Other info: We want to migrate from Cypress to Playwright. For evaluation we created a playright config, and made a pretty basic test of our application. The test runs fine in headless mode, but if we try to start the UI, we get a blank screen and a 404 Error in the browser console. There seems to be a script missing(look at images attached). We don't really have any clue how to fix it or what the problem is.

### Source code

- [ ] I provided exact source code that allows reproducing the issue locally.

**Link to the GitHub repository with the repro**

[https://github.com/your_profile/playwright_issue_title]

or

**Config file**

```js
import type { PlaywrightTestConfig } from '@playwright/test';
import { devices } from '@playwright/test';

const config: PlaywrightTestConfig = {
  testDir: './integration/playwright/tests',
  testMatch: /.*\.ts/,
  timeout: 30 * 1000,
  expect: {
    timeout: 5000,
  },
  fullyParallel: true,
  forbidOnly: !!process.env.CI,
  retries: process.env.CI ? 2 : 0,
  workers: process.env.CI ? 1 : undefined,
  reporter: 'html',
  use: {
    actionTimeout: 0,
    trace: 'on-first-retry',
    baseURL: 'http://localhost:4200/xxx',
  },
  projects: [
    {
      name: 'chromium',
      use: {
        ...devices['Desktop Chrome'],
      },
    },

    {
      name: 'firefox',
      use: {
        ...devices['Desktop Firefox'],
      },
    },

    {
      name: 'webkit',
      use: {
        ...devices['Desktop Safari'],
      },
    },
  webServer: [
    {
      command: 'yarn serve:bootRun checkout',
      url: 'http://localhost:4200/xxx',
      timeout: 120 * 1000,
      reuseExistingServer: true,
    },
    {
      command: 'yarn start:bootRunTesting xxx',
      port: 8080,
      timeout: 120 * 1000,
      reuseExistingServer: true,
    },
    {
      command: 'yarn start:wiremock xxx',
      port: 1213,
      timeout: 120 * 1000,
      reuseExistingServer: true,
    },
  ],
};

export default config;

```

**Test file (self-contained)**

**Steps**
- start the playwright-UI with yarn playwright test --ui
- wait for start
- see blank screen

**Expected**

I see the normal Playwright UI

**Actual**

I see a blank screen and an error in the browser console.

![image](https://github.com/microsoft/playwright/assets/58068363/111b45cd-f33a-4158-b546-221fd1a69e7f)
<img width="1283" alt="image" src="https://github.com/microsoft/playwright/assets/58068363/b54aeda4-99a4-4a3a-a744-448f86a3b569">
<img width="971" alt="image" src="https://github.com/microsoft/playwright/assets/58068363/4ee4c264-5533-4ec0-acea-77294dd3934f">

**Comment by pavelfeldman** [comment]
It is hard to tell what is wrong w/o a local repro. Try bisecting your config (remove those webServers for example) to see it is helps.  Also, do tests work for you? (without --ui)

**Comment by Chrisman94** [comment]
as i said, the tests run fine in headless mode. 
i already removed the webservers from the config, but the problem stays.

**Comment by pavelfeldman** [comment]
I see. I don't think we'll be able to act on this issue since we can't reproduce it and you seem to be the only one affected. Please let us know if you figure it out and there is something we can do on our side.

**Comment by Chrisman94** [comment]
if i setup playwright out of the scope(parent folder) of package.json and tsconfig.json, the UI is starting. Your answer is kind of disappointing. Could you at least give me some input, what is the purpose of this wsport.js file, and what could be the reason that it isn't loaded or in this case is not found. This should be something playwright specific, because it belongs to the UI I guess.

**Comment by pavelfeldman** [comment]
Yeah, sorry about that. Usually we would require a repro we can repeat locally to be able to look into the issue. wsport is just one of the files that our UI mode is using (it is a web app under the hood), so not locating it suggests that something is unusual with either installation or your system.

**Comment by pavelfeldman** [comment]
We need more information to act on this report. As long as we can't repro it, it is unlikely with can make progress with it. Allow me to close it, but please file a new one and link to this issue when you get back to it!

**Comment by RodrigoWebDev** [comment]
Same thing here. Tests work fine, but blank screen in UI mode. I'm running the example scenario that were included when installing it:

<img width="960" alt="image" src="https://github.com/microsoft/playwright/assets/30677819/08cc21d8-1983-4f94-aa87-773a3ef46131">

**Comment by MehmetCanBOZ** [comment]
I would like to draw attention to a particular observation: when executing tests in UI mode and selecting "run" for either the entire test file or a describe block, the test passes, yet the UI mode displays a blank page. However, when running the test for a specific case, the actual screen is visible. I am seeking clarification on this discrepancy and expressing a desire to have visibility into all tests in the life-cycle UI mode, rather than exclusively for individual tests. Your insights into the underlying reason for this behavior would be greatly appreciated.

**Run for spec file**
![Screenshot 2024-02-08 at 14 50 20](https://github.com/microsoft/playwright/assets/66769855/31aac23e-1006-4d04-9804-196e0c438336)


Run for only one specific test
![Screenshot 2024-02-08 at 14 50 28](https://github.com/microsoft/playwright/assets/66769855/bc4558a3-1a20-4e70-bc00-4a161831cd27)

**Comment by lukosevicius** [comment]
Why was this closed? The issue persists

**Comment by mswezey23** [comment]
I fixed it be going back from v1.42 to v1.41. Not the ideal fix, but unblocks me. 

I was running into this issue as well.

**Comment by calebsmithdev** [comment]
Chiming in that I had the same issue on 1.43.0. I downgraded to 1.41 and my UI is working fine again.

**Comment by Etheryte** [comment]
@pavelfeldman I think this issue should be reopened, it's clearly an ongoing problem in both 1.42 and 1.43 that isn't a one-off setup issue. We're running into the same problem on 1.43, downgrading to 1.41 fixes the issue.

**Comment by Jordan466** [comment]
Remember to `await` the `async` APIs. I forgot to do this and I noticed the UI fails to load the page if you don't do this.

**Comment by elf-ioki** [comment]
> I would like to draw attention to a particular observation: when executing tests in UI mode and selecting "run" for either the entire test file or a describe block, the test passes, yet the UI mode displays a blank page. However, when running the test for a specific case, the actual screen is visible. 

I can confirm this observation. When I fully expand all the tests and run them individually, the actual screen is visible.
<img width="249" alt="Screenshot 2024-05-03 at 14 27 38" src="https://github.com/microsoft/playwright/assets/112640038/b5d7d495-538e-4869-a61f-a786e5e57a8d">

**Comment by tigerabrodi** [comment]
@Etheryte 

Encountering it too in 1.43, downgraded to 1.41, still encountering it. 😕 

The test runs and passes, but while it runs and during pause it shows a blank page.

**Comment by arthur1993-code** [comment]
@pavelfeldman The issue is also acutal on 
Playwright Version: v1.44
Operating System: macOS 14.4.1 
Browser: All

In my case, when i run in ui mode first time, it shows blank page. If i run the test on a ui mode window again, it starts to show everything. So the issue is only on first run of the ui mode.

**Comment by OleJoik** [comment]
I am also seeing this issue in 1.41.

![image](https://github.com/microsoft/playwright/assets/57186239/a54e5be7-e965-4057-bce7-828fb3df492f)

For me, no rerunning of the tests resolves the issue, the preview is always blank, however, default dummy tests pass.

Tests are ran against playwright server running in docker as such:
```
docker run -p 3005:3005 --rm --init -it mcr.microsoft.com/playwright:v1.41.0-jammy /bin/sh -c "cd /home/pwuser && npx -y playwright@1.41.0 run-server --port 3005 --host 0.0.0.0"
Listening on ws://0.0.0.0:3005/
```

... And UI:

```
PW_TEST_CONNECT_WS_ENDPOINT=ws://127.0.0.1:3005/ npx playwright test --ui-port=3000
```

System is arch linux
browsers: all
playwright versions: attempted with 1.41 and 1.44 for both docker and locally

**Comment by jrasanen** [comment]
I can't get the UI show up with any browser engine. Playwright's XHR console has a lot of failing requests to URL `http://[::1]:50603/trace/snapshot/undefined?trace=..trace url..`.

**Comment by Armadillidiid** [comment]
I'm on version `1.45.2`  using the same jammy docker image version and experiencing this issue. This issue should please be reopened

**Comment by sizyrg** [comment]
1.45.2 also has this issue where it stopped saving snapshot after 30 seconds of test, everything after that is not saved, extending the test timeout in config or in particular test with test.setTimeout doesn't change anything, it just shows about: blank

**Comment by kresimir-coko** [comment]
I had this issue when using `test.step`, weird idk

**Comment by anthony-bernardo** [comment]
Same, why this issue is closed ?

**Comment by jrasanen** [comment]
> Same, why this issue is closed ?

@anthony-bernardo it was already answered: missing repro. I don't think this will be opened, unless someone creates a repository with a working reproduction.

**Comment by animencoding** [comment]
facing same issue inside of docker only

**Comment by IsmailSAly** [comment]
Facing same issue on 1.48 and 1.49.0 and 1.49.1. Problem only happens on Webkit with device emulation. 
If i don't set viewport page shows up but freezes.

**Comment by RambokDev** [comment]
Same issue using playwright in docker, got blank screen.

**Comment by EndrewSouza** [comment]
2025 and still facing the same problem

**Comment by Yohannes-Asfaw** [comment]
i am still facing the same issue with Version 1.52.0

**Comment by Nvveen** [comment]
I think I have the same issue. Adding my first specs to login from a login page to the dashboard, and up until the `expect` the URL is correct and the dashboard shows, but the page goes blank on the expect line. I am also running from a docker container as a Remote Devcontainer from VSCode in Windows 11. Also get a lot of the same errors as mentioned in this thread. I can sometimes fix the visibility by adding a timeout to the page or whatever, but that messes up the tests themselves. Mousing over the 'after hooks' and the page is visible again, although I still see a lot of loading spinners (specific to our dashboard), even though I can see the requests being completed. I have tried waiting for network requests, but that doesn't seem to do anything as far as I can tell.

Edit: Hovering after the after hooks doesn't always show it. If I run the test multiple times it seems to stop showing it as much. Switching browsers doesn't really matter here.

Edit 2: Only adding a manual timeout of 1000ms seems to consistenly allow me to see the page.

Edit 3: After much testing, I think I've gotten a better idea of what is going wrong. 
Something else that happened, that seemed to be a separate issue at first, was that my Angular app with our own table component using structural directives, wasn't being rendered properly. Structural directives were not being applied, but their literal contents were being projected as the contents of the table component ignoring the table component's own contents. This has never happened in any browser in any environment before. 
If I run the tests in headed mode from the container browser, it launches it (I think via X forwarding or something?) and the tests run normally and everything looks right.
If I start in UI mode, even from the container browser, it looks wrong, same as if I open UI mode in a normal browser tab.
If I take screenshots during tests, those look normal too.
This makes me think that the inline view of the app inside UI mode improperly renders or takes snapshots and that is where the visually broken behavior shows, and why in some steps of the test  the window doesn't show anything at all. In UI mode I can also open that frame in a new window, and instead of showing a normal URL in the URL bar, it shows something like `[::1]:41234/trace/snapshot.html?r=http...` for the screens that have content, and for the end step of where I don't see anything it can't actually find the snapshot.html file.

**Comment by drvic10k** [comment]
I had the same problem and the root cause was that I was starting the tests from root folder and not from the project folder in terminal
when I navigate to the project folder before executing  `npx playwright test --ui` it wors fine

**Comment by Nvveen** [comment]
Yes, my issue with the traces is a separate issue at this point I think. I can't make a reproduction though to submit a separate bug report.

**Comment by Sir-hennihau** [comment]
I noticed that some failed assertions give a blank screen and others give a ui preview even if it failed

f.e. `  await expect(page).toHaveTitle(/bad title/);` as a failed test always gives me a page afterwards to look at

while   `await page.getByText("bad text").click();` will always result in a a white screen

<img width="1285" alt="Image" src="https://github.com/user-attachments/assets/bf07d4a6-1889-4b88-a9a8-22330bd7e737" />

**Comment by callen1994** [comment]
I'm encountering the same issue

**Comment by efstajas** [comment]
I also have & have always had this issue with Playwright on both Mac OS and Ubuntu. I noticed today that passing the `--headed` flag to the playwright command fixes it entirely for me, but of course you're going to see the actual browser windows too.

**Comment by sashokbg** [comment]
Here are my two cents:

I am using archlinux (not supported system). When I try to install the playwright browsers I get a warning "BEWARE: your OS is not officially supported by Playwright; downloading fallback build for ubuntu20.04-x64."
It turns out that this chrome built for ubuntu does not play nice under my archlinux, which explains the blank pages.

I ended up tricking playwright to use my already installed chromium by doing the following:

- Remove caches:
`rm -rf  ~/.cache/ms-playwright` (or wherever the cache is on your system)

- Try to run playwright and it will give an error similar to:
`Error: browserType.launch: Executable doesn't exist at /home/alexander/.cache/ms-playwright/chromium-1187/chrome-linux/chrome`

- `mkdir -p /home/alexander/.cache/ms-playwright/chromium-1187/chrome-linux/`
- Get your current browsers bin location 'which chromium'
- `ln -s /usr/bin/chromium /home/alexander/.cache/ms-playwright/chromium-1187/chrome-linux/chrome`

Now it works with my system's browser, and the pages are all fine

**Comment by prachisarkarNL** [comment]
I ran this command -- npm run download-chromium and after that all screenshots were visible.

**Comment by nzx-Lewis** [comment]
I was having this issue using a remotely hosted devcontainer, I can only have MS edge installed on my work machine but I was able to see the window contents when I opened via the vscode "Simple browser: show" command
